flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosApplicationMasterRunner.java

```diff
@@ -0,0 +1,618 @@
+				config, mesosConfig, ENV,
+				taskManagerParameters, taskManagerConfig,
+				workingDir, getTaskManagerClass(), artifactServer, LOG);
+
+			// ----------------- (4) start the actors -------------------
+
+			// 1) JobManager & Archive (in non-HA case, the leader service takes this)
+			// 2) Web Monitor (we need its port to register)
+			// 3) Resource Master for Mesos
+			// 4) Process reapers for the JobManager and Resource Master
+
+			// 1: the JobManager
+			LOG.debug("Starting JobManager actor");
+
+			// we start the JobManager with its standard name
+			ActorRef jobManager = JobManager.startJobManagerActors(
+				config, actorSystem,
+				new scala.Some<>(JobManager.JOB_MANAGER_NAME()),
+				scala.Option.<String>empty(),
+				getJobManagerClass(),
+				getArchivistClass())._1();
+
+
+			// 2: the web monitor
+			LOG.debug("Starting Web Frontend");
+
+			webMonitor = BootstrapTools.startWebMonitorIfConfigured(config, actorSystem, jobManager, LOG);
+			if(webMonitor != null) {
+				final URL webMonitorURL = new URL("http", appMasterHostname, webMonitor.getServerPort(), "/");
+				mesosConfig.frameworkInfo().setWebuiUrl(webMonitorURL.toExternalForm());
+			}
+
+			// 3: Flink's Mesos ResourceManager
+			LOG.debug("Starting Mesos Flink Resource Manager");
+
+			// create the worker store to persist task information across restarts
+			MesosWorkerStore workerStore = createWorkerStore(config);
+
+			// we need the leader retrieval service here to be informed of new
+			// leader session IDs, even though there can be only one leader ever
+			LeaderRetrievalService leaderRetriever =
+				LeaderRetrievalUtils.createLeaderRetrievalService(config, jobManager);
+
+			Props resourceMasterProps = MesosFlinkResourceManager.createActorProps(
+				getResourceManagerClass(),
+				config,
+				mesosConfig,
+				workerStore,
+				leaderRetriever,
+				taskManagerParameters,
+				taskManagerContext,
+				numInitialTaskManagers,
+				LOG);
+
+			ActorRef resourceMaster = actorSystem.actorOf(resourceMasterProps, "Mesos_Resource_Master");
+
+
+			// 4: Process reapers
+			// The process reapers ensure that upon unexpected actor death, the process exits
+			// and does not stay lingering around unresponsive
+
+			LOG.debug("Starting process reapers for JobManager");
+
+			actorSystem.actorOf(
+				Props.create(ProcessReaper.class, resourceMaster, LOG, ACTOR_DIED_EXIT_CODE),
+				"Mesos_Resource_Master_Process_Reaper");
+
+			actorSystem.actorOf(
+				Props.create(ProcessReaper.class, jobManager, LOG, ACTOR_DIED_EXIT_CODE),
+				"JobManager_Process_Reaper");
+		}
+		catch (Throwable t) {
+			// make sure that everything whatever ends up in the log
+			LOG.error("Mesos JobManager initialization failed", t);
+
+			if (actorSystem != null) {
+				try {
+					actorSystem.shutdown();
+				} catch (Throwable tt) {
+					LOG.error("Error shutting down actor system", tt);
+				}
+			}
+
+			if (webMonitor != null) {
```

Comment:
Interesting; this was copied from YarnApplicationMasterRunner.  I will fix both.